packages/shared/src/encryption/shared-secret.ts

```diff
@@ -0,0 +1,44 @@
+import { chacha20poly1305 } from '@noble/ciphers/chacha';
+import { randomBytes } from '@noble/hashes/utils';
+
+import {
+  DecryptionError,
+  DecryptionErrorCode,
+  EncryptionError,
+  EncryptionErrorCode,
+} from './errors';
+
+export function encryptWithSharedSecret(
+  sharedSecret: Uint8Array,
+  plaintext: Uint8Array,
+): Uint8Array {
+  const nonce = randomBytes(12); // Generate a 12-byte nonce
```

Comment:
Is there a const for `12` somewhere, or one that can be created?

Reply:
Sure! The TS code is updated accordingly now.

(There is no constant for the original Rust code: `pub type Nonce = GenericArray<u8, U12>;`. So, if you think so, an issue at nucypher-core could be created to suggest having this as somehow a constant)

Reply:
Creating a simple constant like you did in the ts file is sufficient. Thanks!